bin/tfidf_search.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import os
+import sqlite3
+import pandas as pd
+from sklearn.metrics.pairwise import linear_kernel
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+
+"""
+This script takes a zettelkasten note filename and sorts the available notes by their similarity
+using Term Frequency-Inverse Document Frequency (TF-IDF). The idea is to use this to help one make
+connections in their zettelkasten they may have not thought of initially. 
+
+I'm not sure how well this works in practice yet since my zettelkasten is just starting to fill up.
+It does seem to find notes I would say are generally related. I have limited the search to print the
+top 20 matches.
+
+I currently utilize sirupsen's search tool to use this (https://github.com/sirupsen/zk/blob/master/bin/zks). 
+I have a second search function for when I want to try TF-IDF that puts 'python --filename "zk filename.md" |' in
+front of the fzf. This will give you a view to scroll through related files with the same opening and linking commands
+as zk-search. 
+"""
+
+
+def vectorize_text(series):
+    """
+    Uses sklearn text vectorizer. Will do the necessary preprocessing like removing stop words,
+    transform to lowercase etc.
+
+    :param series: Pandas Series object
+    :return: matrix of tf-idf features
+    """
+    return TfidfVectorizer().fit_transform(series.values)
```

Comment:
Presumably this is the most expensive operation. This takes about ~1s on my machine with roughly 1,000 notes. That's OK for me, but if you wanted to nerd out, I think you can use [`fts4aux`](https://www.sqlite.org/fts3.html#fts4aux) to get the raw tokenization from Sqlite's FTS to avoid computing it again.

Reply:
Yeah definitely was a next step for me to try to make this a bit faster. There are a few things we can store so they don't have to be recomputed. I'll probably need to get lower-level and program the TFIDF preprocessing stuff myself instead of using the sklearn vectorizer (which is fine just will take a bit).

Reply:
Ya, not a requirement to get this in